networkit/cpp/embedding/BiasedRandomWalk.cpp

```diff
@@ -34,10 +34,28 @@ BiasedRandomWalk::BiasedRandomWalk(const Graph *graph)
     // pre-allocate unordered maps:
     graph->forNodes([&](node v) {
         auto degv = graph->degreeOut(v);
-        for (auto n : graph->neighborRange(v)) {
-            // init index2node:
-            index2node[v].push_back(n);
-            graphData->data[v][n] = AliasSampler(degv);
+        if (graph->isDirected()) {
+            for (auto n : graph->neighborRange(v)) {
+                // local numbering of neighbors by index in index2node[v]:
+                index2node[v].push_back(n);
+            }
```

Comment:
Reserving space should improve performance. We could also use neighbor iterators to avoid the loop.
```suggestion
            // local numbering of neighbors by index in index2node[v]:
            index2node[v].reserve(graph->degree(v));
            index2node[v].insert(index2node[v].begin(), graph->neighborRange(v).begin(), graph->neighborRange(v).end());
```